akka-http/src/main/scala/akka/http/scaladsl/server/Rejection.scala

```diff
@@ -92,6 +94,15 @@ final case class MalformedHeaderRejection(headerName: String, errorMsg: String,
   extends jserver.MalformedHeaderRejection with RejectionWithOptionalCause
 
 /**
+ * Rejection created by [[akka.http.scaladsl.server.directives.HeaderDirectives.checkSameOrigin]].
+ * Signals that the request was rejected because `Origin` header value is invalid.
+ */
+final case class InvalidOriginHeaderRejection(invalidOrigins: immutable.Seq[SHttpOrigin])
```

Comment:
One more thought: I wonder if it would be better to report allowed origins here. Like we do for `Accept` and `Accept-Encoding`. (So the client knows what to do instead of what's forbidden.)

I'd also change to simply `InvalidOriginRejection` as suggested previously.